blob/service.go

```diff
@@ -193,8 +193,7 @@ func (s *Service) getByCommitment(
 
 	namespacedShares, err := s.shareGetter.GetSharesByNamespace(ctx, header.DAH, nID)
 	if err != nil {
-		if errors.Is(err, share.ErrNamespaceNotFound) ||
-			errors.Is(err, share.ErrNotFound) {
+		if errors.Is(err, share.ErrNotFound) {
 			err = ErrBlobNotFound
```

Comment:
Should we also short-circuit here for when len of namespacedShares == 0?

Reply:
It depends on BlobModule `ErrBlobNotFound` definition. I assumed that it is indicating that node was not able to find data, which means user should retry. I think it would be nice to document `ErrBlobNotFound` so it is clear what it actually means. 

More detailed answer to your question. When namespacedShares == 0 it could be 2 cases:
1. `share.ErrNotFound` (node wasn't able to find data, retry needed)
2. namespace not included (no retry needed)

Reply:
`share.ErrNotFound` will be gone soon: https://github.com/celestiaorg/celestia-node/issues/2386